content/master/concepts/composition-functions.md

```diff
@@ -1,1026 +1,637 @@
-      timeout: 30s
-    # An arbitrary Kubernetes resource. Passed to the Function as the config
-    # block of its FunctionIO. Doesn't need to exist as a Custom Resource (CR),
-    # since this resource doesn't exist by itself in the API server but must be
-    # a valid Kubernetes resource (have an apiVersion and kind).
-    config:
-      apiVersion: database.example.org/v1alpha1
-      kind: Config
-      metadata:
-        name: cloudsql
-      spec:
-        version: POSTGRES_9_6
+      - name: storage-bucket
+        base:
+          apiVersion: s3.aws.upbound.io/v1beta1
+          kind: Bucket
+        patches:
+        - type: FromCompositeFieldPath
+          fromFieldPath: spec.bucketRegion
+          toFieldPath: spec.forProvider.region
 ```
-{{< /expand >}}
 
-Use `kubectl describe <xr-kind> <xr-name>` to debug Composition Functions. Look
-for status conditions and events. Most Functions will emit events associated
-with the XR if they experience issues.
+The `functions.yaml` file contains the Functions the Composition references in
+its pipeline steps:
 
-## Building a function
+```yaml
+---
+apiVersion: pkg.crossplane.io/v1beta1
+kind: Function
+metadata:
+  name: function-patch-and-transform
+spec:
+  package: xpkg.upbound.io/crossplane-contrib/function-patch-and-transform:v0.1.4
+```
+{{</expand>}}
 
- Crossplane doesn't have opinions about how a Composition Function is
- implemented. Functions must:
+`crossplane beta render` prints resources as YAML to stdout. It prints the
+composite resource first, followed by the resources the composition functions
+created.
 
- * Be packaged as an OCI image, where the `ENTRYPOINT` is the Function.
- * Accept input in the form of a `FunctionIO` document on stdin.
- * Return the `FunctionIO` they were passed, optionally mutated, on stdout.
- * Run within the constraints specified by the Composition that includes them,
-   such as timeouts, compute, network access.
+```yaml
+---
+apiVersion: example.crossplane.io/v1
+kind: XBucket
+metadata:
+  name: example-render
+---
+apiVersion: s3.aws.upbound.io/v1beta1
+kind: Bucket
+metadata:
+  annotations:
+    crossplane.io/composition-resource-name: storage-bucket
+  generateName: example-render-
+  labels:
+    crossplane.io/composite: example-render
+  ownerReferences:
+  - apiVersion: example.crossplane.io/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: XBucket
+    name: example-render
+    uid: ""
+spec:
+  forProvider:
+    region: us-east-2
+```
 
-This means Functions may be written using a general-purpose programming language
-like Python, Go, or TypeScript. They may also be implemented using a shell
-script, or an existing tool like Helm or Kustomize.
+{{<hint "important">}}
+Running `crossplane beta render` requires [Docker](https://www.docker.com).
 
-### FunctionIO
+See the [Crossplane CLI docs](https://github.com/crossplane/docs/pull/584) to
```

Comment:
This link is a placeholder. Needs updating when https://github.com/crossplane/docs/pull/584 is merged.